code/modules/mining/remote_fob/remote_fob_computer.dm

```diff
@@ -0,0 +1,199 @@
+
+/////////////////////////////// the camera computer
+
+/obj/machinery/computer/camera_advanced/remote_fob
+	name = "FOB Construction Drone Control"
+	desc = "A computer console equipped with camera screen and controls for a planetside deployed construction drone. Materials or equipment vouchers can be added simply by inserting them into the computer."
+	icon = 'icons/Marine/remotefob.dmi'
+	icon_state = "fobpc"
+	req_access = list(ACCESS_MARINE_REMOTEBUILD)
+	networks = FALSE
+	off_action = new/datum/action/innate/camera_off/remote_fob
+	jump_action = null
+	var/drone_creation_allowed = TRUE
+	var/obj/docking_port/stationary/marine_dropship/spawn_spot
+	var/lz
+	var/datum/action/innate/remote_fob/metal_cade/metal_cade
+	var/max_metal = 50 //mostly to prevent jokers collecting all the metal and dumping it in
+	var/metal_remaining = 0
+	var/datum/action/innate/remote_fob/plast_cade/plast_cade
+	var/max_plasteel = 50
+	var/plasteel_remaining = 0
+	var/datum/action/innate/remote_fob/toggle_wiring/toggle_wiring //whether or not new barricades will be wired
+	var/do_wiring = FALSE
+	var/datum/action/innate/remote_fob/sentry/sentry
+	var/sentry_remaining = 0
+	var/datum/action/innate/remote_fob/eject_metal/eject_metal
+	var/datum/action/innate/remote_fob/eject_plasteel/eject_plasteel
+
+/obj/machinery/computer/camera_advanced/remote_fob/Initialize()
+	. = ..()
+	metal_cade = new()
+	plast_cade = new()
+	toggle_wiring = new()
+	sentry = new()
+	eject_metal = new()
+	eject_plasteel = new()
+
+	RegisterSignal(SSdcs, COMSIG_GLOB_DROPSHIP_TRANSIT, .proc/disable_drone_creation)
+
+/obj/machinery/computer/camera_advanced/remote_fob/proc/disable_drone_creation()
+	drone_creation_allowed = FALSE
+	UnregisterSignal(SSdcs, COMSIG_GLOB_DROPSHIP_TRANSIT)
+
+
+/obj/machinery/computer/camera_advanced/remote_fob/Destroy()
+	drone_creation_allowed = null
+	spawn_spot = null
+	lz = null
+	metal_cade = null
```

Comment:
```suggestion
	QDEL_NULL(metal_cade)
```
References to things that are made by the object and should be gone with it need to be qdeleted as well.
Same for the other ones created on Initialize

Reply:
Can you elaborate further? Just for my understanding.

Reply:
Of course.
The reason we null things on `Destroy` is to remove references to other existing objects so the GC processes things correctly.
Values like numbers or strings are not references to anything, so it doesn't make a difference if they get nulled.
Things we create on Initialize and that should be destroyed along with the thing that created should be deleted on the creator's `Destroy` proc, besides nulled. hence the `QDEL_NULL`